### PR TITLE
Load and select optimization profiles for static TRT engine

### PR DESCRIPTION
**Pull request by tfeher**
This PR implements TRT optimization profile handling in case of static engine:
- Initializes the optimization profile helper class from ICudaEngine
- Select the optimization profile for the input shape

This is PR is necessary to use dynamic shape mode with static engines.

Tracker #52012

Tagging @bixia1 for review.

### CODE REVIEW

**Comment by tfeher** [comment]
Improved trt_engine_op_test to cover the static engine path (including profile handling).

The logic of handling and creating engines in `TrtEngineOp::GetEngine` needs some simplification. I propose to do that in a follow up PR,  so that here we have a clear overview of the changes.

@bixia1 ready for review

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc [hunk 1]
```diff
@@ -153,6 +185,24 @@ class TRTEngineOpTestBase : public OpsTestBase {
...
+ public:
+  TRTEngineOpTestWithParam() : param_(GetParam()) {}
+
+ protected:
+  TestParam param_;
+};
+
+const absl::string_view TRTEngineOpTestBase::kOpName = "myop";
+
+constexpr std::array<TestParam, 2> TestParameters{TestParam{false, false},
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc:200]: s/false, false/false, true/?

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc [hunk 2]
```diff
@@ -82,6 +88,32 @@ class TRTEngineOpTestBase : public OpsTestBase {
...
+      params.use_implicit_batch = use_implicit_batch;
+      params.trt_logger_name = "DefaultLogger";
+
+      TrtShapeOptimizationProfile profile;
+      TensorShape my_shape;
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{4, 2}, &my_shape));
+      profile.AddShape({my_shape, {}});
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{1, 2}, &my_shape));
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc:108]: Currently, we only have tests with input shape {1,2}. For this reason, shall we remove {4,2}?
Also, shall we also add a test that has an input shape rather then these shapes used to prebuild the engine?

Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc:109]: We should not remove the unused profile. It is there to test profile selection: we have {4,2} as profile zero, to run the test TRTEngineOp has to select profile 1. I have added a comment explaining that.

Good point about testing with an input for which we do not have an engine. I assume that native segment fall-back is the expected behavior. I have corrected GetEngine and added a test.

File: tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc (hunk 1 above)
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op_test.cc]: We have two parameters one with `static_engine=false`, and another with `static_engine=true`. The first value `implicit_batch` is always false in the current tests, therefore I have removed it from TestParams (for now).

**Review by tfeher — COMMENTED** [review]
Thansk @bixia1 for the comments, I have fixed the issues!

**Review by bixia1 — APPROVED** [review]

**Pull request merged**